save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/usersettings/UserSettingsOrganizationsMenuView.kt

```diff
@@ -51,16 +76,50 @@ class UserSettingsOrganizationsMenuView : UserSettingsView() {
                                 }
                             }
                             div {
-                                className = ClassName("mr-3")
-                                val role = state.userInfo?.name?.let {
-                                    organizationDto.userRoles[it]
-                                } ?: Role.NONE
-                                +role.formattedName
+                                className = ClassName("col-5 align-self-right d-flex align-items-center justify-content-end")
+                                val role = state.userInfo?.name?.let { organizationDto.userRoles[it] } ?: Role.NONE
+                                div {
+                                    className = ClassName("mr-3")
+                                    +role.formattedName
+                                }
+                                if (role.isHigherOrEqualThan(Role.OWNER)) {
+                                    div {
+                                        button {
+                                            className = ClassName("btn mr-3")
+                                            fontAwesomeIcon(icon = faTrashAlt)
+                                            id = "remove-organization-${organizationDto.name}"
+                                            onClick = {
+                                                setDeleteOrganization(organizationDto)
+                                                windowOpenness.openWindow()
+                                            }
+                                        }
+                                    }
+                                }
                             }
                         }
                     }
                 }
             }
         }
     }
+
+    private fun deleteOrganization(organizationDto: OrganizationDto) {
+        val headers = jsonHeaders
+        lateinit var responseFromDeleteOrganization: Response
+        scope.launch {
+            responseFromDeleteOrganization =
+                    delete(
+                        "$apiUrl/organizations/${organizationDto.name}/delete",
+                        headers,
+                        body = undefined,
+                        loadingHandler = ::noopLoadingHandler,
+                    )
+        }.invokeOnCompletion {
+            if (responseFromDeleteOrganization.ok) {
+                setState {
+                    window.location.reload()
+                }
+            }
```

Comment:
What is going on in this lines?

Reply:
You can do something like:
_selfOrganizationDtos = state.selfOrganizationDtos.minusElement(organizationDto)_